src/commonControls/numberControl/NumberControlAPL.ts

```diff
@@ -0,0 +1,167 @@
+        },
+    };
+
+    /**
+     * The APL dataSource to use when requesting a number value
+     *
+     */
+    export function numberPadDataSourceGenerator() {
+        return (act: RequestValueAct | RequestChangedValueAct) => {
+            return {
+                numPadData: {
+                    controlId: act.control.id,
+                    headerTitle: i18next.t('NUMBER_CONTROL_DEFAULT_APL_HEADER_TITLE'),
+                    isValidValue: (act.control as NumberControl).state.isValidValue,
+                    validationFailedMessage: (act.control as NumberControl).getAplValidationFailedMessage(),
+                },
+            };
+        };
+    }
+
+    /**
+     * The APL document to use when requesting a number value.
+     * For information on EditText, see:
+     * https://developer.amazon.com/en-US/docs/alexa/alexa-presentation-language/apl-edittext.html
+     *
+     */
+    export function numberPadDocumentGenerator() {
+        return {
+            type: 'APL',
+            version: '1.4',
+            import: [
+                {
+                    name: 'alexa-layouts',
+                    version: '1.2.0',
+                },
+            ],
+            styles: {
+                EditStyle: {
+                    values: [
+                        {
+                            borderWidth: 2,
+                            borderStrokeWidth: 1,
+                            borderColor: 'darkgrey',
+                            hintColor: 'grey',
+                            hintWeight: '200',
+                            fontSize: '20dp',
+                        },
+                        {
+                            when: '${state.focused}',
+                            borderColor: 'green',
+                            borderStrokeWidth: 2,
+                        },
+                    ],
+                },
+            },
+            mainTemplate: {
+                parameters: ['numPadData'],
+                items: {
+                    type: 'Container',
+                    width: '100vw',
+                    height: '100vh',
+                    direction: 'column',
+                    items: [
+                        {
+                            type: 'AlexaBackground',
+                        },
+                        {
+                            type: 'AlexaHeader',
+                            headerTitle: '${numPadData.headerTitle}',
+                            headerDivider: true,
+                            theme: 'dark',
+                        },
+                        {
+                            type: 'Container',
+                            height: '75%',
+                            alignItems: 'center',
+                            justifyContent: 'center',
+                            items: [
+                                {
+                                    type: 'EditText',
+                                    when: '${environment.aplVersion != null}',
```

Comment:
Interesting... so if there is trouble with APL, the user will see the question and can reply with voice?  Our other option is to check the APL version in the control when you check if APL interface is present.  Something to discuss.

Reply:
I actually misunderstood the docs previously, `environment.aplVersion` will only return null in the case of APL 1.0. This is anyway not required since without the 'when', the component won't show anyway if the device's APL version doesn't support the feature (in this case, < 1.4). I'll remove it.

More here: https://developer.amazon.com/en-US/docs/alexa/alexa-presentation-language/apl-latest-version.html#apl-version-checking

And yes, we're showing just the question (and the validation failed message if applicable) if there are version issues.